apiserver/facades/agent/controllercharm/register.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package controllercharm
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/juju/errors"
+	"github.com/juju/names/v4"
+
+	apiservererrors "github.com/juju/juju/apiserver/errors"
+	"github.com/juju/juju/apiserver/facade"
+	"github.com/juju/juju/environs/bootstrap"
+	"github.com/juju/juju/state"
+)
+
+// Register is called to expose a package of facades onto a given registry.
+func Register(registry facade.FacadeRegistry) {
+	registry.MustRegister("ControllerCharm", 1, func(ctx facade.Context) (facade.Facade, error) {
+		return newAPI(ctx)
+	}, reflect.TypeOf((*API)(nil)))
+}
+
+// newAPI creates a new server-side controllercharm API facade.
+func newAPI(ctx facade.Context) (*API, error) {
+	model, err := ctx.State().Model()
```

Comment:
We need to ensure this facade can be called by the controller charm from the controller model ONLY.